rules/falco_rules.yaml

```diff
@@ -1659,6 +1792,57 @@
   priority: NOTICE
   tags: [network, mitre_exfiltration]
 
+# When filled in, this should look something like:
+# (proc.env contains "HTTP_PROXY=http://my.http.proxy.com ")
+# The trailing space is intentional so avoid matching on prefixes of
+# the actual proxy.
+- macro: allowed_ssh_proxy_env
+  condition: (always_true)
+
+- rule: Program run with disallowed http proxy env
+  desc: An attempt to run a program with a disallowed HTTP_PROXY environment variable
+  condition: >
+    spawned_process and not allowed_ssh_proxy_env and proc.env icontains HTTP_PROXY
```

Comment:
Should we add a list of commands like `curl` and `wget` in this case?

Reply:
I couldn't think of a way to get a good list though, and I didn't want to limit it to only curl/wget.

Reply:
then this will be a little bit noisy, as not all the commands in the shell are relevant to the PROXY env variable. We can have a list and add more later.

Reply:
Yeah I struggled between looking for the execs (bad when lots of execs pass around the bad env) and the connects (bad when there are a lot of connects). So I guess I'll add a list.